invokeai/backend/model_management/model_cache.py

```diff
@@ -487,7 +493,20 @@ def _make_cache_room(self, model_size):
             else:
                 pos += 1
 
-        gc.collect()
+        if models_cleared < 0:
```

Comment:
When is this true? models_cleared is only ever incremented

Reply:
Good catch 😓 . Should be `> 0`. I guess this just emphasizes that this is really a _guess_ as to how often gc.collect() should be called.

I'll put up a PR with a fix. And then hopefully community testing on a range of systems will tell us if this condition is appropriate.

Reply:
PR to patch: https://github.com/invoke-ai/InvokeAI/pull/5036#issue-1977322287